docs/development/quickstart.rst

```diff
@@ -0,0 +1,365 @@
+is likely that one or more of those CI checks will fail when you make a pull request,
+resulting in lost time (yours and CI resources).
+
+Installation is straightforward. From the root of the astropy repository, run::
+
+    pre-commit install
+
+Now all of the styling checks will be
+run each time you commit changes without your needing to run each one manually.
+
+.. tip:: To learn more about pre-commit, see the :ref:`pre-commit` section.
+
+.. _contributing.pull_request:
+
+Creating and submitting a pull request
+======================================
+
+You can contribute bug fixes, new features, and documentation updates by submitting a
+GitHub pull request (PR). This section will guide you through the process. We encourage
+you to `ask for help <https://www.astropy.org/help.html>`_ if you get stuck. The astropy
+community is welcoming and friendly and will help you!
+
+.. _contributing.version_control:
+
+GitHub and Git
+--------------
+
+Astropy is hosted on `GitHub <https://www.github.com/astropy/astropy>`_, and to
+contribute, you will need to sign up for a `free GitHub account
+<https://github.com/signup/free>`_.
+
+We use `Git <https://git-scm.com/>`_ for version control and to allow many people to
+work together on the project. If you are new to Git then take a moment to look at the
+:ref:`git-resources` page.
+
+If you are new to contributing to projects through forking on GitHub, take a look at the
+`GitHub documentation for contributing to projects
+<https://docs.github.com/en/get-started/quickstart/contributing-to-projects>`_. GitHub
+provides a quick tutorial using a test repository that may help you become more familiar
+with forking a repository, cloning a fork, creating a feature branch, pushing changes
+and making pull requests.
+
+Creating a feature branch
+-------------------------
+
+Your local ``main`` branch should always reflect the current state of astropy repository.
+First ensure it's up-to-date with the main astropy repository::
+
+    git checkout main
+    git pull upstream main --ff-only
+
+Now create a feature branch for making your changes. For example::
+
+    git switch -c subpackage-new-feature
+
+This changes your working branch from ``main`` to the ``subpackage-new-feature`` branch.
+Keep any changes in this branch specific to one bug or feature so it is clear what the
+branch brings to astropy. You can have many feature branches and switch in between them
+using the ``git switch`` command.
+
+Using a descriptive branch name can help you stay organized. For example
+```io-ascii-commented-header-15513``` might be a good name for a branch that fixes the
+commented header issue `#15513 <https://github.com/astropy/astropy/issues/15513>`_ in
+the ``io.ascii`` sub-package.
+
+.. Important:: Never use the ``main`` branch for making changes. Always create a new
+   feature branch for your changes.
+
+When you want to update the feature branch with changes in main after
```

Comment:
This feels slightly out of place here, and you have a more useful warning further down. So perhaps remove here? Or move the important point here.